internal/container/reference.go

```diff
@@ -19,29 +19,55 @@ type RefSet struct {
 	// The image name (minus the Tilt tag) as referenced from the cluster (in k8s YAML,
 	// etc.) (Often this will be the same as the LocalRef, but in some cases they diverge:
 	// e.g. when using a local registry with KIND, the image localhost:1234/my-image
-	// (LocalRef) is referenced in the YAML as http://registry/my-image (ClusterRef).
-	ClusterRef reference.Named
+	// (LocalRef) is referenced in the YAML as http://registry/my-image (clusterRef).
+	// clusterRef is optional; if not provided, we assume LocalRef == clusterRef.
+	clusterRef reference.Named
 }
 
-// SimpleRefSet makes a ref set for the given selector, assuming that LocalRef
-// and ClusterRef are equal.
+func NewRefSet(confRef RefSelector, localRef, clusterRef reference.Named) RefSet {
+	return RefSet{
+		ConfigurationRef: confRef,
+		LocalRef:         localRef,
+		clusterRef:       clusterRef,
```

Comment:
seems fine, tho this makes me wonder if you should store the localRegistry / clusterRegistry instead of the localRef / clusterRef.

that way, if you can't update the LocalRef and forget to update the clusterRef

Reply:
oooh I like that!